### ISSUE TITLE
assets.get method throw `Error: no such type: google.ads.googleads.v8.errors.GoogleAdsFailure`

### ISSUE THREAD

**Issue by arditti** [Opteo/google-ads-api#309]
Hey,

Im using version 8.0.1 of this package, and encounter with a new error/behaviour:
i ran this code several times, the exact same code, with the exact same input: (the ids been removed for obvious reason)
```
await customer.assets.get('customers/<CUSTOMER_ID>/assets/<ASSET_ID>');
```
and half of the times i get the following error:
```
/Users/user/dev/project-service/node_modules/protobufjs/src/namespace.js:383
        throw Error("no such type: " + path);
              ^
Error: no such type: google.ads.googleads.v8.errors.GoogleAdsFailure
    at Root.lookupType (/Users/user/dev/project-service/node_modules/protobufjs/src/namespace.js:383:15)
    at GoogleErrorDecoder.decodeProtobufAny (/Users/user/dev/project-service/node_modules/google-gax/src/googleError.ts:67:28)
    at /Users/user/dev/project-service/node_modules/google-gax/src/googleError.ts:112:14
    at Array.map (<anonymous>)
    at /Users/user/dev/project-service/node_modules/google-gax/src/googleError.ts:111:57
    at Array.forEach (<anonymous>)
    at GoogleErrorDecoder.decodeRpcStatusDetails (/Users/user/dev/project-service/node_modules/google-gax/src/googleError.ts:106:15)
    at Immediate.callback (/Users/user/dev/project-service/node_modules/google-gax/src/call.ts:123:39)
    at processImmediate (node:internal/timers:466:21)
    at process.topLevelDomainCallback (node:domain:147:15)
    at process.callbackTrampoline (node:internal/async_hooks:128:24)
```

Even when i try to wrap this piece of code with try/catch block - it doesn't seems to help, and my entire process crash.

reverting back to version 5.2.0 didn't helped.

Please advise.

**Comment by MitulGedeeya** [comment]
@arditti @WillCooter  I am also facing this issue since 3 to 4 hours and checking for the solution. I have raised same error here

https://github.com/Opteo/google-ads-api/issues/308

can you please update us on it?

**Comment by arditti** [comment]
Happens on other method as well, like getting reports for `ad_group_ad` resources.
Im pretty afraid that try/catch does catch these errors

**Comment by rynti** [comment]
This broke my build as well, I was able to fix it by pinning the `google-gax` package to version `2.19.0`.

(I did it by specifying `"google-gax": "2.19.0",` in the package.json and, since I'm using yarn, I ran `npx yarn-deduplicate` to make sure only one google-gax version is used in the yarn lockfile)

**Comment by MitulGedeeya** [comment]
@rynti I tried this solution but getting same error. :(

**Comment by MitulGedeeya** [comment]
It seems it is due to following changes in google-gax.

![image](https://user-images.githubusercontent.com/6191937/128297149-9f483a07-71be-427d-9545-60a2fe4fd48d.png)

**Comment by rynti** [comment]
> @rynti I tried this solution but getting same error. :(

Sorry to hear that, are you sure you are only having google-gax installed once, and at the correct version (I tested with 2.19.0)? With yarn it's easy to check (by investigating the `yarn.lock`), not sure how to test this with regular NPM though, I guess the NPM lockfile is similar, or worst case you can look manually inside the node_modules folders and sub-folders.

Because when I changed my package.json, by default yarn installed google-gax twice, once at 2.19.0 because it was requested in my package.json, but then a different version within the module tree for `google-ads-api` and `google-ads-node`. That's why I used `npx yarn-deduplicate` to make sure only one google-gax version is installed.

**Comment by kritzware** [comment]
Thanks for all the info everyone. This is indeed not good as it completely kills the process, even with error handling. I was hoping this PR https://github.com/Opteo/google-ads-api/pull/311 might fix the issue as it updated `google-gax` in both repos, but it doesn't seem to solve it.

I'm going to do a few more checks to ensure it's not on our end and then I'll raise the issue today with our Google contacts.

**Comment by MitulGedeeya** [comment]
We are facing serious issues of restarts of the node app on GAE due to this issue. can anyone suggest temporary workaround for it?

**Comment by MitulGedeeya** [comment]
@arditti have you done anything related to it?

**Comment by arditti** [comment]
> @arditti have you done anything related to it?

Sadly we didn't find any workaround yet. we are looking for any clue in order to solve it. 

will keep this thread updated on any findings.

**Comment by MitulGedeeya** [comment]
Thanks. I am also finding a workaround for it. I will also keep thread updated.

**Comment by alexander-fenster** [comment]
This should be fixed in a `google-gax` dependency (in v2.22.0) that was just released, so please reinstall the dependencies and make sure `npm ls google-gax` shows 2.22.0 or newer.

**Comment by MitulGedeeya** [comment]
Thanks @alexander-fenster . It seems it fixed an issue. we can see the original error in our catch blocks

**Comment by kritzware** [comment]
This issue should now be resolved in `8.0.2` of this library (contains a `google-ads-node` update including the above `google-gax` fix) – thanks @alexander-fenster 👍

**Comment by arditti** [comment]
> This issue should now be resolved in `8.0.2` of this library (contains a `google-ads-node` update including the above `google-gax` fix) – thanks @alexander-fenster 👍

Many thanks for the help on this. sadly since this update we faced with a new issue:
https://github.com/Opteo/google-ads-api/issues/312
